### ISSUE TITLE
`/savings/requestSizingV2` API error when partial data is available.

### ISSUE THREAD

**Issue by jcharcalla** [kubecost/cost-analyzer-helm-chart#2374]
This was reported by a customer, I have not reproduced it.

After recently enabling the [containerStats pipeline ](https://docs.kubecost.com/using-kubecost/navigating-the-kubecost-ui/savings/auto-request-sizing/containerstats-pipeline) they found that only API requests with a window shorter than the duration which it had been running were successful. For example, the container stats pipeline was enabled 4 days ago, a window of 3 days works as expected, however a window of 7 days produced the following error.

```
{"code":2,"message":"Could not retrieve CPU recommendation: Recommending requests from data: CPU algorithm set to quantile but no quantile data is available\n"}*%*
```

Redacted version of the API request.

```
curl -v -G \
-d 'algorithmCPU=quantile' \
-d 'qCPU=.95' \
-d 'targetCPUUtilization=0.8' \
-d 'window=7d' \
-d 'minRecCPUMillicores=75' \
--data-urlencode 'filter=annotation[foo1]:"dev1"+annotation[foo2]:"example"' \
-H "cookie: service-session=${SSO_SESSION}" \
${KUBECOST_ADDRESS}/savings/requestSizingV2
```

As discussed with @michaelmdresser this is probably a bug because data is available during the 7 day window as shown with a 3 day window. 

> If metrics were missing, then underlying data (Allocation and ContainerStats) will be built from what is available. Request sizing recommendations are derived from that data. "no ... data is available" should only happen if there is truly no data in KC for the given window; if that error is occurring and there is data then its a bug

**Comment by thomasvn** [comment]
Thanks @jcharcalla! I've logged this as a bug and we'll try to fix in next release.

**Comment by michaelmdresser** [comment]
https://kubecost.atlassian.net/browse/BURNDOWN-154

**Comment by mmurph3** [comment]
@thomasvn their current workaround is to use quantile but they'd like to use max.

**Comment by michaelmdresser** [comment]
@mmurph3 wrong issue?

**Comment by nickcurie** [comment]
As far as I understand it, containerStats breaks down a given window into 24 hour chunks (i.e. `window=7d` is converted to 7 windows of 1d). If there's a 1d window in those 7 days that doesn't have any data, this error should be thrown. 

I've tried to repro this on my own cluster and I'm not getting any error when querying a window that would span before containerStats was enabled. I enabled containerStats ~3 hours ago, and am able to query a 7d window without issue. 

I think it would help if we were able to run this same API request for windows of `4d` `5d` `6d` to determine where the missing data might be. 
Debug logs might also help here. If not all coverage is available for containerStats, logs indicating such would be present. We should either see `Have <num of unavailable> windows to build` if there are still some unbuilt windows, or `All coverage available. Waiting a bit.` if the containerStats pipeline finished building all the windows.
